lib/mix/tasks/git_hooks/run.ex

```diff
@@ -17,6 +17,8 @@ defmodule Mix.Tasks.GitHooks.Run do
   You can also all the hooks which are configured with `mix git_hooks.run all`.
   """
 
+  @requirements ["app.start"]
```

Comment:
I was not aware of this annotation ([for reference](https://github.com/elixir-lang/elixir/issues/9786)). I'm still not sure how this is needed, but in any case, shouldn't be `@requirements ["compile"]` instead?

Reply:
Hey @qgadrian , I just pushed a new commit that changes the current behavior for MFAs a little bit.

Basically I removed the `@requirements ["app.start"]` and replaced it with an `on-demand` load of the app.

The logic will try to run the MFA as before the first time (so, in case the MFA is from a dependency, this will work just fine and it will be way faster since there is no need to load the app at all), if the MFA run fails with a `UndefinedFunctionError`, then the code will load the full app with `Mix.Task.run("app.start")` and re-run the MFA code.

In my tests this is working great and, as long as the hook code is in a dependency, not the main project, it will run pretty fast.

What do you think?

Reply:
@sezaru what about always starting the app in that case? If it is already running it will be a `:noop`, so it should be deterministic.
But in any case, I will give it a try as I haven't set up any example project yet to check the MFA hooks configuration.